dynawo/sources/Models/CPP/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.h

```diff
@@ -0,0 +1,410 @@
+  */
+  void defineElements(std::vector<Element> &elements, std::map<std::string, int >& mapElement);
+  /**
+  * @brief ModelLoadRestorativeWithLimits discrete variables evaluation
+  *
+  * Get the discrete variables' value depending on current simulation instant and
+  * current state variables values.
+  *
+  * @param t Simulation instant
+  */
+  void evalZ(const double & t);
+
+ private:
+  /**
+  * @brief compute value
+  * @return value
+  */
+  double UfRaw() const;
+  /**
+  * @brief compute value
+  * @return value
+  */
+  double Uf() const;
+  /**
+  * @brief compute value
+  * @return value
+  */
+  double UfRawPrim() const;
+  /**
+  * @brief get real value of voltage
+  * @return value
+  */
+  double Ur() const;
+  /**
+  * @brief get imaginary value of voltage
+  * @return value
+  */
+  double Ui() const;
+  /**
+  * @brief get real value of current
+  * @return value
+  */
+  double Ir() const;
+  /**
+  * @brief get imaginary value of current
+  * @return value
+  */
+  double Ii() const;
+  /**
+  * @brief compute value
+  * @return value
+  */
+  double U() const;
+  /**
+  * @brief compute value
+  * @return value
+  */
+  double U2() const;
+  /**
+  * @brief get the active power in p.u. (SNREF)
+  * @return value
+  */
+  double P() const;
+  /**
+  * @brief get the reactive power in p.u. (SNREF)
+  * @return value
+  */
+  double Q() const;
+  /**
+  * @brief calculated value
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double P_dUr(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief calculated value
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double P_dUi(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief calculated value
+  * @param U voltage
+  * @return value
+  */
+  double P_dUf(const double& U) const;
+  /**
+  * @brief calculated value
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double Q_dUr(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief calculated value
+  * @param ur real part of the voltage
+  * @param ui imaginary part of the voltage
+  * @param U voltage
+  * @param U2 voltage square
+  * @return value
+  */
+  double Q_dUi(const double& ur, const double& ui, const double& U, const double& U2) const;
+  /**
+  * @brief calculated value
+  * @param U voltage
+  * @return value
+  */
+  double Q_dUf(const double& U) const;
+
+  State connectionState_;  ///< "internal" load connection status
+  State connectionState_prev;  ///< "internal" load connection status at previous timestamp
```

Comment:
typo: timestep

Reply:
from google:  A timestamp marks when something happened; a time step is the interval between two times.
I'd let timestamp as it indicates state at t-1